smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/SymbolVisitor.java

```diff
@@ -235,9 +237,24 @@ public Symbol operationShape(OperationShape shape) {
     @Override
     public Symbol stringShape(StringShape shape) {
         // Enums that provide a name for each variant create an actual enum type.
-        return shape.getTrait(EnumTrait.class)
-                .map(enumTrait -> createEnumSymbol(shape, enumTrait))
-                .orElseGet(() -> createSymbolBuilder(shape, "string").build());
+        Optional<EnumTrait> enumTrait = shape.getTrait(EnumTrait.class);
+        if (enumTrait.isPresent()) {
+            return createEnumSymbol(shape, enumTrait.get());
+        }
+
+        // Handle media type generation, defaulting to a string.
+        Optional<MediaTypeTrait> mediaTypeTrait = shape.getTrait(MediaTypeTrait.class);
+        if (mediaTypeTrait.isPresent()) {
+            String mediaType = mediaTypeTrait.get().getValue();
+            switch (mediaType) {
+                case "application/json":
```

Comment:
- [ ] You could also use the wrapper for any media type that ends with "+json": https://tools.ietf.org/html/rfc6839#section-3.1
- [ ] It looks like this check if performed in a couple places. If we make this work with "+json", then an "isJsonMediaType" utility method or something similar should probably be added.
- [ ] I think this should have a test too.